### PR TITLE
Update service resource

### LINKED ISSUE #17: Update service resource
Add docs, add unit tests, make matchers, make it a bit more InSpec-y.

### PR DESCRIPTION
**Pull request by clintoncwolfe**
Closes #17 - see it for details

### CODE REVIEW

File: docs/resources/habitat_service.md [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+```ruby
+describe habitat_service(origin: 'core', name: 'httpd') do
+  its('dependency_names') { should include 'core/pcre' }
+  its('dependency_names') { should_not include 'core/nginx' }
+end
+```
+
+### dependency_ids
+
+Array of Strings. A list of the packages that this service depends on, in the form `dep_origin/dep_name/1.2.3/20190325123456`. Note that version strings are up to the maintainers of the individual projects, and need not be of the form `1.2.3`
```
Comment (mjingle) [inline docs/resources/habitat_service.md]: Second sentence, "Note that version strings are" 
In parsing this sentence, is it meant to be "Note that in this version, strings are"? Or are "version strings" something else?

File: docs/resources/habitat_service.md [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+
+## Matchers
+
+Use [Matchers](https://www.inspec.io/docs/reference/glossary/#matcher) to create tests that test a true or false question.
+
+InSpec includes a number of universal matchers(https://www.inspec.io/docs/reference/matchers/).  See below for matchers specific to this resource.
+
+### have_standalone_topology
+
+True if the service is configured in a [`standalone`](https://www.habitat.sh/docs/using-habitat/#standalone) topology.
```
Comment (mjingle) [inline docs/resources/habitat_service.md]: Would this section of matchers have "True" as `true` formatting-wise?

Comment (mjingle) [inline docs/resources/habitat_service.md]: Also, would it be more clear to have the sentence format of each matcher statement begin with "Matcher returns `true` if..."?

File: docs/resources/habitat_service.md (hunk 1 above)
Comment (clintoncwolfe) [reply docs/resources/habitat_service.md]: Yes, this need clarification -thanks. A version string is the third component, here `1.2.3`. The temptation is assume it is always three numbers and two dots, and usually is, but the maker of the software can (and do) do anything they like. So, it's dangerous to make comparisons against that value, unless you know the specific format. used by specific software.

File: docs/resources/habitat_service.md (hunk 2 above)
Comment (clintoncwolfe) [reply docs/resources/habitat_service.md]: Yes, that would be more consistent. The literal value is `true`, and the type is `TrueClass`; "True" is just conversational.

Comment (clintoncwolfe) [reply docs/resources/habitat_service.md]: Sure!

**Pull request merged**